ethcore/src/snapshot/service.rs

```diff
@@ -424,37 +449,102 @@ impl Service {
 			}
 		}
 
+		{
+			*self.status.lock() = RestorationStatus::Initializing {
```

Comment:
nit: I don't think scope here is necessary since `lock()` is not left hand side of the assignment, so guard here is temporary ([play example](http://play.rust-lang.org/?gist=5885b84ebcd4740aaacc5616dcc19146&version=stable&mode=debug))